cpp/test/Slice/print/AllTests.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) ZeroC, Inc.
+
+#include "Ice/Ice.h"
+#include "Test.h"
+#include "TestHelper.h"
+
+#include <iomanip>
+
+using namespace std;
+using namespace Test;
+
+void
+Test::Neighbor::ice_print(ostream& os) const
+{
+    os << name;
+    if (age)
+    {
+        os << " (" << *age << ")";
+    }
+    os << " @ " << address;
+}
+
+template<class T>
+void
+testPrint(const T& value, const string& expected, std::function<void(ostream&)> manip = nullptr)
+{
+    ostringstream os;
+    if (manip)
+    {
+        manip(os);
+    }
+    os << value;
+    string result = os.str();
+    if (result != expected)
+    {
+        cout << "expected: " << expected << endl;
+        cout << "got:      " << result << endl;
+        test(false);
+    }
+}
+
+void
+testSimpleStruct()
+{
+    SimpleStruct simpleStruct{true, 199, 201, 150000, -100000000, 3.14f, 152853.5047l, "hello", FlagColor::Blue};
+    testPrint(
+        simpleStruct,
+        "Test::SimpleStruct{myBool = true, myByte = 199, myShort = 201, myInt = 150000, myLong = -100000000, myFloat = "
+        "3.14, myDouble = 152854, myString = hello, myEnum = 2}");
+
+    // Increasing the floating-point precision to 9 digits changes the output (C++ default is 6).
+    testPrint(
+        simpleStruct,
+        "Test::SimpleStruct{myBool = true, myByte = 199, myShort = 201, myInt = 150000, myLong = -100000000, myFloat = "
+        "3.1400001, myDouble = 152853.505, myString = hello, myEnum = 2}",
+        [](ostream& os) { os << setprecision(9); });
+}
+
+void
+testByteBoolStruct()
+{
+    vector<byte> byteSeq{byte{100}, byte{150}, byte{255}};
+    vector<bool> boolSeq{true, false, true};
+    ByteBoolStruct byteBoolStruct{byteSeq, boolSeq};
+
+    testPrint(byteBoolStruct, "Test::ByteBoolStruct{myByteSeq = [100, 150, 255], myBoolSeq = [true, false, true]}");
+
+    testPrint(
+        byteBoolStruct,
+        "Test::ByteBoolStruct{myByteSeq = [0x64, 0x96, 0xff], myBoolSeq = [true, false, true]}",
+        [](ostream& os) { os << hex << showbase; });
```

Comment:
The user can easily change the formatting. The Ice "print" functions don't change the format flags of the stream.